### PR TITLE
task/ICL-258 get published offchain claims

### PR DESCRIPTION
**Pull request by JGiter**
### Description

https://energyweb.atlassian.net/browse/ICL-258

### CODE REVIEW

File: src/modules/claims/claims.service.ts [hunk 1]
```diff
@@ -590,14 +594,24 @@ export class ClaimsService {
...
-  }: { did?: string } | undefined = {}) {
+  }: { did?: string } | undefined = {}): Promise<Claim[]> {
     const { service } = (await this._didRegistry.getDidDocument({ did })) || {};
-    return service;
+    const issuedClaims = await this.getClaimsBySubject({
+      did,
+      isAccepted: true,
+    });
+    return issuedClaims.filter((c) =>
+      service.some(
```
Comment (Harasz) [inline src/modules/claims/claims.service.ts:609]: This logic is a little bit fuzzy for me. Could you explain what you are trying to achieve here?

Comment (JGiter) [reply src/modules/claims/claims.service.ts:609]: Sure. At the moment getUserClaims actually returns service endpoints. So this change is aimed to return published issued claims, which is ensured by checking that for issued offchain claim exists corresponding service. I would also change method name to getOffchainClaims({isApproved, isPublished}) or even better extend getClaimsBy.. with registrationTypes parameter if this wasn't breaking change affecting a lot of code

Comment (Harasz) [reply src/modules/claims/claims.service.ts]: Firstly, `service` variable can be undefined, so `service.some()` can throw an error.
And what about something like this
 ```suggestion
      const { service } = (await this._didRegistry.getDidDocument({ did })) || {};

if (!service) return []

const issuedClaims = await this.getClaimsBySubject({
  did,
  isAccepted: true,
});

const issuedClaimsTypes = issuedClaims.map(({ claimType }) => claimType);

return issuedClaims.filter(
  ({ registrationTypes, claimType }) =>
    registrationTypes.includes(RegistrationTypes.OffChain) &&
    issuedClaimsTypes.includes(claimType)
);
```

I think it's the code is more readable

Comment (JGiter) [reply src/modules/claims/claims.service.ts]: I am agree with null checking of `service`, but filtering of issued claims is unclear to me. It doesn't seems that filtering excludes not-published claims (not added to document). Also second condition seems like holds true for all claims

Comment (Harasz) [reply src/modules/claims/claims.service.ts]: I agree, `issuedClaimsTypes` should map a claim type from `service` array, then everything should be fine, what you think?

Comment (JGiter) [reply src/modules/claims/claims.service.ts]: Seems fine to me

**Review by Harasz — APPROVED** [review]

**Review by Harasz — APPROVED** [review]

**Pull request merged**